### ISSUE TITLE
Slicing, split and similar tensor modifications

### ISSUE THREAD

**Issue by derpda** [Xilinx/brevitas#351]
First of all, thank you for providing this library and its source code!

I'm running into trouble with slicing and splitting of QuantTensors.
Looking through the examples and the source code, I found no clean solution for my use case.

In the regular pytorch code I'm doing things like the following in a few places
```python
x = x[:, :, :residual.size(-1)]
// ...
xa, xb = x.split(x.size(splitdim) // 2, dim=splitdim)
```
The only way I found that kind of works is 
```python
x = x.set(value=x.value[:, :, :residual.size(-1)])
```
for the first equation, and I guess I could do something similar for the split operation.

However, this seems like a hack to me.
Is this intended usage, am I missing something or is this kind of operation not (yet) supported in Brevitas?

**Comment by volcacius** [comment]
Hello,

Yes there are definitely various tensor manipulation operation missing at the moment. I added a few more recently in the dev branch but many are still missing. 
One issue is that until recently I had a requirement on supporting PyTorch 1.1.0+, and custom tensor-like datatypes are supported only starting from 1.5.0+, so QuantTensor was mainly there to correctly propagate gradients among learned bit-widths with certain ops, but was not a first class citizen. 
The other issue is that Brevitas is a research library first, so it tries to be the least prescriptive possible in order to allow the flexibility to explore new concepts. The downside is that because there are few constraints, the semantics of certain operations can be kind of fuzzy. Example, depending on how you slice a tensor you might also need to slice its scale/zero-point/bit-width, and there are a lot of options there in terms of how they might look like. So overall I prefer to assume that the user knows best and leave it to them to implement it, like you are doing here. 
In the future I'd like to have better structure around custom quantized tensor datatypes, but I'm not there yet.

**Comment by derpda** [comment]
Thank you for your answer. 
I took some more time to study the source code (found a potential bug as well, but I'll make a separate issue for that) and I think I will work around this problem until `QuantTensor` is a child class of `torch.tensor`.
I'll probably keep `return_quant_tensor=False` and rather use input quantization.

Luckily what I am trying to do is not that complicated (I think).
Maybe you can tell me if this would be an appropriate use of the library:
##### What I have:
  - A trained network
  - Statistics about the weights, biases, inputs, outputs
  - A working FPGA implementation in pure HLS (no Vitis-AI etc)
  - Post-training rudimentary fixed-point quantization (`ap_(u)fixed<>`)

##### What I want to do:
  - Improve precision with QAT fine-tuning
  - Keep `ap_fixed`, so probably (?) power-of-two scales and `zero_point=0` everywhere
 
As far as I can tell, all layers and activation functions I need are available in Brevitas.
I am making progress on this, but currently I am having trouble with the bias quantization.
